---
title: "Itch DEA"
author: "Erik Larsen"
date: "2025-02-10"
output: html_document
---

# Overview

This markdown was developed to formally "modernize" the differential expression data analysis of bulk RNA-seq data
for **Dr. Martha Bhattacharya's lab** at the **University of Arizona** for the [Itch paper](https://journals.lww.com/pain/abstract/2022/05000/transmembrane_protein_tmem184b_is_necessary_for.18.aspx)

## Set up Environment {.tabset .tabset-pills .tabset-fade}

#### **Load packages**

```{r Package Load, include = T, message = F, warning = F, echo = T}
packages <- c('BiocManager',
              'tidyverse',
              'stringr',
              'GO.db',
              'org.Mm.eg.db',
              'forcats',
              'ggplot2',
              'ggrepel',
              'ggbiplot',
              'pheatmap')
for ( package in packages ) {
  if (!require(package, character.only = T)) {
    install.packages(package)
    library(package, character.only = T)
  }
}
```

# Read in and Prep aDRG Data {.tabset .tabset-pills .tabset-fade}

#### **Read in aDRG Data**

Not shown; variable names:
+ `aDRG` = adult mouse DRG DEA results
  + `GO_{category}_results` = gene.ontology gene set enrichment analysis results

```{r Read in aDRG RNAseq Results, include = T, message = F, warning = F, echo = F}
aDRG = read.csv("path/to/DESeq2 Expression Results.csv")
aTPM <- read.csv(
  "path/to/DESeq2_Normalized_Counts_Entire_Timecourse_by_Genotype.csv",
  header = T
  ) |>
  dplyr::rename(GeneID = `X`) %>%
  dplyr::rename_with(~gsub(.x,
                           pattern = '\\s|\\.(T|t)rim',
                           replacement = '')) %>%
  dplyr::rename_with(~gsub(.x,
                           pattern = '^(mut[0-9])$',
                           replacement = 'a\\1')) %>%
  dplyr::rename_with(~gsub(.x,
                           pattern = '^(wt[0-9])$',
                           replacement = 'a\\1')) |>
  dplyr::relocate(awt1, .after = p10wtE) |>
  dplyr::relocate(awt3, .before = e13mu1) |>
  dplyr::relocate(awt4, .before = e13mu1) |>
  dplyr::relocate(awt8, .before = e13mu1) |>
  dplyr::relocate(contains('amut'), .after = p10mu4) %>%
  dplyr::rename_with(~gsub(.x,
                           pattern = 'amut([0-9])',
                           replacement = 'amu\\1')) |>
  dplyr::select(c(GeneID, dplyr::starts_with('a')))
```


# Plot aDRG Results

## **DEA**

### **Volcano**

```{r Plot aDRG DEA results, include = T, message = F, warning = F, echo = T,fig.width=12}
  ## create the handy notin pipe operator
"%notin%" = Negate("%in%")

  ## list the differentially expressed genes known to be involved in itch
itch_DEGs = c("Il31ra", "Cysltr2", "Npy2r", "Sst",
              "Htr1a", "P2rx3", "Lpar3", "Lpar5",
              "Scn11a", "Scn10a", "Mrgprd", "Trpc6",
              "Trpc3", "F2rl2", "Htr1f", "Osmr",
              "Fxyd2", "Htr4", "Mrgprx1", "Ptgdr",
              "Trpa1", "Trpm6", "Hrh1", "Mrgpra3", "Tmem184b")

  ## Create grouping and labeling variables
aDRG <- aDRG |> 
  dplyr::mutate(goi = case_when(-log10(AdjP) >= -log10(0.05) &
                                  GeneID %in% itch_DEGs ~ 'Itch-related DEG',
                                -log10(AdjP) >= -log10(0.05) &
                                  GeneID %notin% itch_DEGs ~ 'DEG',
                                TRUE ~ 'Non-DEG'),
                labs = case_when(GeneID %in% c('Tmem184b',
                                               'Il31ra',
                                               'Cysltr2',
                                               'Npy2r',
                                               'Sst',
                                               'Htr1a',
                                               'Htr1f',
                                               'Lpar5',
                                               'Lpar3',
                                               'Mrgprd',
                                               'Mgrpra3',
                                               'Trpa1') ~ GeneID,
                                 TRUE ~ ''))

  ## volcano plot
ggplot(data = aDRG) +
  geom_point(data = subset(aDRG, goi == 'Non-DEG'),
             aes(x = log2.FC.,
                 y = -log10(AdjP),
                 color = goi),
             size = 3,
             alpha = 0.1) +
  geom_point(data = subset(aDRG, goi == 'DEG'),
             aes(x = log2.FC.,
                 y = -log10(AdjP),
                 color = goi),
             size = 4,
             alpha = 0.5) +
  geom_point(data = subset(aDRG, goi == 'Itch-related DEG'),
             aes(x = log2.FC.,
                 y = -log10(AdjP),
                 color = goi),
             size = 5,
             alpha = 1) +
  coord_cartesian(ylim = c(0,53), xlim = c(-3.2,2)) +
  theme_classic() +
  theme(plot.title = element_text(face = 'bold', size = 15),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.ticks = element_line(linetype = 'solid'),
        legend.position = 'none'
        ) +
  labs(title = expression(bold(paste("DEA of ",
                                     italic("Tmem184b")^bold("GT/GT"),
                                     " DRG mRNA Exp. Changes"))),
       x = bquote(~ log[2] ~ '(fold change mRNA Relative to WT)'),
       y = bquote(~ -log[10] ~ '(Adj.P-value)'),
       col = 'Gene\nData\nType'
       ) +
  geom_hline(yintercept = -log10(0.05),
             linetype = 'dashed',
             color = 'firebrick') +
  scale_color_manual(values = c('darkgoldenrod4', 'navy', 'gray48'))+
  geom_text(aes(x = -1.9, 
                y = 2.5,
                label = 'FDR \U2264 0.05'),
            color = 'firebrick',
            size = 5) + 
  geom_text_repel(data = aDRG,
                  aes(label = aDRG$labs),
                  x = aDRG$log2.FC.,
                  y = -log10(aDRG$AdjP),
                  color = 'black',
                  max.overlaps = Inf,
                  size = 5)
```

## **Trx Profiling**

### **Z-score the Normalized Counts**

#### **Z-score Function**

```{r Create Zscore Function, include = T, message = F, warning = F, echo = F}
## Create a function that determines the Zscores of an expression profile (row Z-scores)
Find_Row_Z <- function(Expression_Profile){
  ## Create dataframes to calculate Z-scores across replicates of each gene.
  ## These will hold the original data frame values until filled in later commands
  ## "MeansAndSDs" will store the means and sds of each gene
  MeansAndSDs = Expression_Profile
  ## "GeneMeans" will store only the means by genes for all replicates
  GeneMeans = Expression_Profile
  ## "GeneSDs" will store only the sds by gene for all replicates
  GeneSDs = Expression_Profile
  ## "GeneZ" will store only the Z-scores, which will be used directly to create the heatmaps
  GeneZ = Expression_Profile
  
  ## Create a new column to fill with the means and standard deviations of each gene's transcript counts per million ("TPM")
  MeansAndSDs$mean = 0
  MeansAndSDs$sd = 0
  
  ## Loop through the dataframe and fill the "mean" and "sd" columns with their appropriate values
  for (i in 1:nrow(MeansAndSDs)){
    MeansAndSDs$mean[i] = (sum(Expression_Profile[i,c(2:ncol(Expression_Profile))])/ncol(Expression_Profile[,c(2:ncol(Expression_Profile))]))
  }
  for (j in 1:nrow(MeansAndSDs)){
    MeansAndSDs$sd[j] = sd(Expression_Profile[j,c(2:ncol(Expression_Profile))], na.rm = TRUE)
  }
  ## Create a dataframe storing the gene-specific mean normalized TPM in all columns/replicates for Z-score calculating
  GeneMeans[,c(2:ncol(Expression_Profile))] = MeansAndSDs$mean
  ## Create a dataframe storing the gene-specific normalized TPM standard deviationsin all columns/replicates for Z-score calculating
  GeneSDs[,c(2:ncol(Expression_Profile))] = MeansAndSDs$sd
  ## Create the Z-score dataframe
  GeneZ[,c(2:ncol(Expression_Profile))] = (Expression_Profile[,c(2:ncol(Expression_Profile))] - GeneMeans[,c(2:ncol(Expression_Profile))])/GeneSDs[,c(2:ncol(Expression_Profile))]
  ## Remove the genes that were detected to have 0 TPMs across all samples
  ## Create a function that evaluates a vector/dataframe's (x's) numerical values. Returns equal length vector with T/F bools.
  ## Subset the dataframe that filters those genes.
  row_has_na = apply(GeneZ, 1, function(x){any(is.na(x))})
  GeneZ = GeneZ[!row_has_na,]
  Z <<- GeneZ
}
```

```{r Pass Profile through Zscore Function, include = T, message = F, warning = F, echo = T}
  ## new dataframe should not contain any NAs in p-value columns
aDRG_filtered <- aDRG |>
  dplyr::filter(!is.na(AdjP)) |> # remove any NAs in p-value columns
  dplyr::filter(!grepl(GeneID, # remove rRNAs, mitochondrial tRNAs, pseudogenes 
                       pattern = 'Rps.+.?$|RP.+.?$|Rpl.+.?$|MRPL.+.?$|Mrpl.+.?$|MRPS.+.?$|Mrps.+.?$|.*Rik.+$|.*Rik$|Gm.+.?$|^[A-Z]+[A-Z].+.?$|^[0-9]+.+.?$|mt.+.?$'))

  ## Rename columns
colnames(aTPM) = c("GeneID",
                   "WT1", "WT2", "WT3", "WT4",
                   "Mut1", "Mut2", "Mut3", "Mut4")

  ## Pass profile through Zscore function
Find_Row_Z(Expression_Profile = aTPM[,c(1:9)])
```

### **PCA**

```{r aDRG PCA, include = T, message = F, warning = F, echo = T, fig.height = 12, fig.width = 10}
  ## Create a covariance matrix of the centered and normalized data
Z_cov_mat = cov(Z[,c(2:9)])
  ## Pass the covariance matrix into R's PCA function to obtain the PCA object

PCA = prcomp(Z_cov_mat, center = FALSE, scale. = FALSE)
summary(PCA)


ggbiplot(PCA, 
         ellipse = FALSE,
         obs.scale = 1,
         var.scale = 1,
         labels = colnames(Z[c(2:9)]),
         groups = c(colnames(Z[c(2:9)])[c(1:4)],
                    colnames(Z[c(2:9)])[c(5:8)]),
         labels.size = 9,
         # point.size = 8,
         # varname.size = 20,
         var.axes = FALSE) +
  labs(title = expression(paste(italic("Tmem184b")^"GT/GT", " aDRG mRNA Exp. Changes PCA"))) +
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5, size = 20),
        legend.position = 'none',
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 15),
        legend.text = element_text(size = 14),
        legend.title = element_blank()) +
  scale_color_manual(values = c(rep("darkgoldenrod4", 4), rep("gray25", 4)))
```

### **Hierarchical Clustering the Z-scores**

```{r ReOrder Trx Profile, include = T, warning = F, message = F, echo = T}
  ## Don't remove non-DEGs
Z <- Z |>
  dplyr::filter(GeneID %in% c(aDRG |> 
                                dplyr::select(GeneID) |>
                                unlist() |>
                                as.character()
                              )
                )

  ## Create a list of gene names ordered by euclidean distance by which to re-order the dataframe
Euclid_dist_order = hclust(dist(Z[,c(2:9)], method = "euclidean"))$order
  ## The names (not the numbers)
Euclid_dist_ord_Genes = c(Z$GeneID[Euclid_dist_order])

  ## Transform again to order by clusters (Euclidean distances)
Za = Z |>
  dplyr::mutate(GeneID = factor(GeneID, levels = Euclid_dist_ord_Genes)) |>
  dplyr::arrange(GeneID)

  ## Find where the "Itch-related DEGs" are in the clustered matrix subsetted to DEGs for the heatmap
itch_DEGs = c("Il31ra", "Cysltr2", "Npy2r", "Sst",
              "Htr1a", "P2rx3", "Lpar3", "Lpar5",
              "Scn11a", "Scn10a", "Mrgprd", "Trpc6",
              "Trpc3", "F2rl2", "Htr1f", "Osmr",
              "Fxyd2", "Htr4", "Mrgprx1", "Ptgdr",
              "Trpa1", "Trpm6", "Hrh1", "Mrgpra3", "Tmem184b")

temp = vector()
for( i in 1:length(itch_DEGs)){
  temp[i] = which(Euclid_dist_ord_Genes == itch_DEGs[i])
}
itch_index = temp
#itch_index

  ## Confirm by indexing; copy and paste to customize the clustered heatmap
#Euclid_dist_ord_Genes[c(itch_index)]
  ## Confirm the location/order of the transformed matrix/dataframe
  ## Should be "Il31ra"
Za[itch_index[1],]
```

### **Plot Heatmap**

+ View full heatmap

```{r Plot aDRG heatmap, include = T, message = F, warning = F, echo = T, fig.width=12}
 ## Visualize the Z-scored, Euclidean-clustered and ordered gene TPMs across replicates with "pheatmap"
pheatmap(mat = Za[,2:9],
         color = colorRampPalette(c("navy", "white", "darkgoldenrod4"))((50)),
         clustsering_distance_rows = "euclidean",
         angle_col = 0,
         cutree_rows = 2,
         treeheight_row = 35,
         treeheight_col = 7,
         show_rownames = F)
```


+ Zoom to cluster near Tmem184b

```{r Plot aDRG Tmem Zoom heatmap, include = T, message = F, warning = F, echo = T, fig.width=12}
  ## Visualize around the Tmem184b cluster
pheatmap(mat = Za[1135:1175,2:9],
         color = colorRampPalette(c("navy", "white", "darkgoldenrod4"))((50)),
         clustsering_distance_rows = "euclidean",
         angle_col = 0,
         treeheight_row = 35,
         treeheight_col = 9,
         labels_row = Euclid_dist_ord_Genes[1135:1175])
```

+ Zoom to itch-related genes

```{r Plot itch DEG heatmap, include = T, message = F, warning = F, echo = T, fig.with = 12}
  ## Visualize the "Itch-related DEGs"
pheatmap(mat = Za[c(itch_index),2:9],
         color = colorRampPalette(c("navy", "white", "darkgoldenrod4"))((50)),
         clustsering_distance_rows = "euclidean",
         angle_col = 0,
         treeheight_row = 35,
         treeheight_col = 9,
         cutree_rows = 4,
         labels_row = Euclid_dist_ord_Genes[c(itch_index)])
```


# Read in and Prep eDRG Data {.tabset .tabset-pills .tabset-fade}

#### **Read in eDRG Data**

Not shown; variable names:
+ `({dev_stage_group})_DRG` = DRG DEA results
  + `({dev_stage_group})_GO_{category}_results` = gene.ontology results

```{r Read in eDRG Results, include = T, warning = F, message = F, echo = F}
e13_DRG <- read.csv("path/to/DESeq2_result_file_on_GT_e13_PARA.csv", header = F)
colnames(e13_DRG) = colnames(aDRG)[c(1:7)]

e13_DRG_GO_BP_results <- read.table(
  'path/to/e13_DRG_GO_analysis_BP.txt',
  sep = '\t',
  skip = 11,
  header = T,
  quote = '') |>
  dplyr::rename_with(~gsub(.x,
                           pattern = '\\.\\.',
                           replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x,
                           pattern = 'Mus.musculus.|upload_1|\\.$',
                           replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.biological.process.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `1411`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8),
                       ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term,
                  into = c("GO_Term", "GO_Term_ID"),
                  sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:',
                                    gsub(GO_Term_ID,
                                         pattern = '\\)',
                                         replacement = ''))) |>
  dplyr::mutate(GO_Type = "BP", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment,
                                              pattern = '[:alnum:]+') |>
                  as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) |> # sort by the highest FE and lowest adjusted p-value
  dplyr::filter(GO_Term == 'neuron differentiation')

eDRG_GO_results <- read.table(
  'path/to/e13_DRG_GO_analysis_BP.txt',
  sep = '\t',
  skip = 11,
  header = T,
  quote = '') |>
  dplyr::rename_with(~gsub(.x,
                           pattern = '\\.\\.',
                           replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x,
                           pattern = 'Mus.musculus.|upload_1|\\.$',
                           replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.biological.process.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `1411`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8),
                       ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term,
                  into = c("GO_Term", "GO_Term_ID"),
                  sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:',
                                    gsub(GO_Term_ID,
                                         pattern = '\\)',
                                         replacement = ''))) |>
  dplyr::mutate(GO_Type = "BP", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment,
                                              pattern = '[:alnum:]+') |>
                  as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) |>
  dplyr::mutate(devStage = 'e13') |>
  
  dplyr::full_join(
    read.table(
      'path/to/p0_DRG_GO_analysis_BP.txt',
      sep = '\t',
      skip = 11,
      header = T,
      quote = '') |>
    dplyr::rename_with(~gsub(.x,
                             pattern = '\\.\\.',
                             replacement = '')) |> # remove the .'s  from the GO BP column
    dplyr::rename_with(~gsub(.x,
                             pattern = 'Mus.musculus.|upload_1|\\.$',
                             replacement = '')) |> # remove excess strings
    dplyr::rename(GO_Term = GO.biological.process.complete) |> # rename the BP column
    dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
    dplyr::rename(Overlap = `3078`) |> # rename the overlap column
    dplyr::mutate(across(c(2:4,7,8),
                         ~as.numeric(.x))) |> # convert to numeric
    tidyr::separate(GO_Term,
                    into = c("GO_Term", "GO_Term_ID"),
                    sep = '\\s\\(GO:') |> # split the Term ID from the Term
    dplyr::mutate(GO_Term_ID = paste0('GO:',
                                      gsub(GO_Term_ID,
                                           pattern = '\\)',
                                           replacement = ''))) |>
    dplyr::mutate(GO_Type = "BP", .after = GO_Term_ID) |> # create the category column
    dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment,
                                                pattern = '[:alnum:]+') |>
                    as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
    dplyr::arrange(FDR, desc(fold.Enrichment)) |>
    dplyr::mutate(devStage = 'p0')
  ) |>
  
  dplyr::full_join(
    read.table(
      'path/to/p10_DRG_GO_analysis_BP.txt',
      sep = '\t',
      skip = 11,
      header = T,
      quote = '') |>
    dplyr::rename_with(~gsub(.x,
                             pattern = '\\.\\.',
                             replacement = '')) |> # remove the .'s  from the GO BP column
    dplyr::rename_with(~gsub(.x,
                             pattern = 'Mus.musculus.|upload_1|\\.$',
                             replacement = '')) |> # remove excess strings
    dplyr::rename(GO_Term = GO.biological.process.complete) |> # rename the BP column
    dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
    dplyr::rename(Overlap = `107`) |> # rename the overlap column
    dplyr::mutate(across(c(2:4,7,8),
                         ~as.numeric(.x))) |> # convert to numeric
    tidyr::separate(GO_Term,
                    into = c("GO_Term", "GO_Term_ID"),
                    sep = '\\s\\(GO:') |> # split the Term ID from the Term
    dplyr::mutate(GO_Term_ID = paste0('GO:',
                                      gsub(GO_Term_ID,
                                           pattern = '\\)',
                                           replacement = ''))) |>
    dplyr::mutate(GO_Type = "BP", .after = GO_Term_ID) |> # create the category column
    dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment,
                                                pattern = '[:alnum:]+') |>
                    as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
    dplyr::arrange(FDR, desc(fold.Enrichment)) |>
    dplyr::mutate(devStage = 'p10')
  ) |> 
  
  dplyr::full_join(
    read.table(
      'path/to/e13_DRG_GO_analysis_CC.txt',
      sep = '\t',
      skip = 11,
      header = T,
      quote = ''
    ) |> 
    dplyr::rename_with(~gsub(.x,
                             pattern = '\\.\\.',
                             replacement = '')) |> # remove the .'s  from the GO CC column
    dplyr::rename_with(~gsub(.x,
                             pattern = 'Mus.musculus.|upload_1|\\.$',
                             replacement = '')) |> # remove excess strings
    dplyr::rename(GO_Term = GO.cellular.component.complete) |> # rename the CC column
    dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
    dplyr::rename(Overlap = `1411`) |> # rename the overlap column
    dplyr::mutate(across(c(2:4,7,8),
                         ~as.numeric(.x))) |> # convert to numeric
    tidyr::separate(GO_Term,
                    into = c("GO_Term", "GO_Term_ID"),
                    sep = '\\s\\(GO:') |> # split the Term ID from the Term
    dplyr::mutate(GO_Term_ID = paste0('GO:',
                                      gsub(GO_Term_ID,
                                           pattern = '\\)',
                                           replacement = ''))) |>
    dplyr::mutate(GO_Type = "CC", .after = GO_Term_ID) |> # create the category column
    dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment,
                                                pattern = '[:alnum:]+') |>
                    as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
    dplyr::arrange(FDR, desc(fold.Enrichment)) |>
    dplyr::mutate(devStage = 'e13')
  ) |>
  
    dplyr::full_join(
    read.table(
      'path/to/p0_DRG_GO_analysis_CC.txt',
      sep = '\t',
      skip = 11,
      header = T,
      quote = ''
    ) |> 
    dplyr::rename_with(~gsub(.x,
                             pattern = '\\.\\.',
                             replacement = '')) |> # remove the .'s  from the GO CC column
    dplyr::rename_with(~gsub(.x,
                             pattern = 'Mus.musculus.|upload_1|\\.$',
                             replacement = '')) |> # remove excess strings
    dplyr::rename(GO_Term = GO.cellular.component.complete) |> # rename the CC column
    dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
    dplyr::rename(Overlap = `3078`) |> # rename the overlap column
    dplyr::mutate(across(c(2:4,7,8),
                         ~as.numeric(.x))) |> # convert to numeric
    tidyr::separate(GO_Term,
                    into = c("GO_Term", "GO_Term_ID"),
                    sep = '\\s\\(GO:') |> # split the Term ID from the Term
    dplyr::mutate(GO_Term_ID = paste0('GO:',
                                      gsub(GO_Term_ID,
                                           pattern = '\\)',
                                           replacement = ''))) |>
    dplyr::mutate(GO_Type = "CC", .after = GO_Term_ID) |> # create the category column
    dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment,
                                                pattern = '[:alnum:]+') |>
                    as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
    dplyr::arrange(FDR, desc(fold.Enrichment)) |>
    dplyr::mutate(devStage = 'p0')
  ) |>
  
    dplyr::full_join(
    read.table(
      'path/to/p10_DRG_GO_analysis_CC.txt',
      sep = '\t',
      skip = 11,
      header = T,
      quote = ''
    ) |> 
    dplyr::rename_with(~gsub(.x,
                             pattern = '\\.\\.',
                             replacement = '')) |> # remove the .'s  from the GO CC column
    dplyr::rename_with(~gsub(.x,
                             pattern = 'Mus.musculus.|upload_1|\\.$',
                             replacement = '')) |> # remove excess strings
    dplyr::rename(GO_Term = GO.cellular.component.complete) |> # rename the CC column
    dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
    dplyr::rename(Overlap = `107`) |> # rename the overlap column
    dplyr::mutate(across(c(2:4,7,8),
                         ~as.numeric(.x))) |> # convert to numeric
    tidyr::separate(GO_Term,
                    into = c("GO_Term", "GO_Term_ID"),
                    sep = '\\s\\(GO:') |> # split the Term ID from the Term
    dplyr::mutate(GO_Term_ID = paste0('GO:',
                                      gsub(GO_Term_ID,
                                           pattern = '\\)',
                                           replacement = ''))) |>
    dplyr::mutate(GO_Type = "CC", .after = GO_Term_ID) |> # create the category column
    dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment,
                                                pattern = '[:alnum:]+') |>
                    as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
    dplyr::arrange(FDR, desc(fold.Enrichment)) |>
    dplyr::mutate(devStage = 'p10')
  ) |>
  
    dplyr::full_join(
    read.table(
      'path/to/e13_DRG_GO_analysis_MF.txt',
      sep = '\t',
      skip = 11,
      header = T,
      quote = ''
    ) |> 
    dplyr::rename_with(~gsub(.x,
                             pattern = '\\.\\.',
                             replacement = '')) |> # remove the .'s  from the GO MF column
    dplyr::rename_with(~gsub(.x,
                             pattern = 'Mus.musculus.|upload_1|\\.$',
                             replacement = '')) |> # remove excess strings
    dplyr::rename(GO_Term = GO.molecular.function.complete) |> # rename the MF column
    dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
    dplyr::rename(Overlap = `1411`) |> # rename the overlap column
    dplyr::mutate(across(c(2:4,7,8),
                         ~as.numeric(.x))) |> # convert to numeric
    tidyr::separate(GO_Term,
                    into = c("GO_Term", "GO_Term_ID"),
                    sep = '\\s\\(GO:') |> # split the Term ID from the Term
    dplyr::mutate(GO_Term_ID = paste0('GO:',
                                      gsub(GO_Term_ID,
                                           pattern = '\\)',
                                           replacement = ''))) |>
    dplyr::mutate(GO_Type = "MF", .after = GO_Term_ID) |> # create the category column
    dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment,
                                                pattern = '[:alnum:]+') |>
                    as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
    dplyr::arrange(FDR, desc(fold.Enrichment)) |>
    dplyr::mutate(devStage = 'e13')
  ) |>
  
    dplyr::full_join(
    read.table(
      'path/to/p0_DRG_GO_analysis_MF.txt',
      sep = '\t',
      skip = 11,
      header = T,
      quote = ''
    ) |> 
    dplyr::rename_with(~gsub(.x,
                             pattern = '\\.\\.',
                             replacement = '')) |> # remove the .'s  from the GO MF column
    dplyr::rename_with(~gsub(.x,
                             pattern = 'Mus.musculus.|upload_1|\\.$',
                             replacement = '')) |> # remove excess strings
    dplyr::rename(GO_Term = GO.molecular.function.complete) |> # rename the MF column
    dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
    dplyr::rename(Overlap = `3078`) |> # rename the overlap column
    dplyr::mutate(across(c(2:4,7,8),
                         ~as.numeric(.x))) |> # convert to numeric
    tidyr::separate(GO_Term,
                    into = c("GO_Term", "GO_Term_ID"),
                    sep = '\\s\\(GO:') |> # split the Term ID from the Term
    dplyr::mutate(GO_Term_ID = paste0('GO:',
                                      gsub(GO_Term_ID,
                                           pattern = '\\)',
                                           replacement = ''))) |>
    dplyr::mutate(GO_Type = "MF", .after = GO_Term_ID) |> # create the category column
    dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment,
                                                pattern = '[:alnum:]+') |>
                    as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
    dplyr::arrange(FDR, desc(fold.Enrichment)) |>
    dplyr::mutate(devStage = 'p0')
  ) |>
  
    dplyr::full_join(
    read.table(
      'path/to/p10_DRG_GO_analysis_MF.txt',
      sep = '\t',
      skip = 11,
      header = T,
      quote = ''
    ) |> 
    dplyr::rename_with(~gsub(.x,
                             pattern = '\\.\\.',
                             replacement = '')) |> # remove the .'s  from the GO MF column
    dplyr::rename_with(~gsub(.x,
                             pattern = 'Mus.musculus.|upload_1|\\.$',
                             replacement = '')) |> # remove excess strings
    dplyr::rename(GO_Term = GO.molecular.function.complete) |> # rename the MF column
    dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
    dplyr::rename(Overlap = `107`) |> # rename the overlap column
    dplyr::mutate(across(c(2:4,7,8),
                         ~as.numeric(.x))) |> # convert to numeric
    tidyr::separate(GO_Term,
                    into = c("GO_Term", "GO_Term_ID"),
                    sep = '\\s\\(GO:') |> # split the Term ID from the Term
    dplyr::mutate(GO_Term_ID = paste0('GO:',
                                      gsub(GO_Term_ID,
                                           pattern = '\\)',
                                           replacement = ''))) |>
    dplyr::mutate(GO_Type = "MF", .after = GO_Term_ID) |> # create the category column
    dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment,
                                                pattern = '[:alnum:]+') |>
                    as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
    dplyr::arrange(FDR, desc(fold.Enrichment)) |>
    dplyr::mutate(devStage = 'p10')
  )


p0_DRG <- read.csv(
  "path/to/DESeq2_result_file_on_GT_p0_mu234_wtBCD_PARA.csv",
  header = T
  )


p10_DRG <- read.csv(
  "path/to/DESeq2_result_file_on_GT_p10_mu123_wt1CE_PARA.csv",
  header = T
  )

TPM <- read.csv(
  "path/to/DESeq2_Normalized_Counts_Entire_Timecourse_by_Genotype.csv",
  header = T
  ) |>
  dplyr::rename(GeneID = `X`) %>%
  dplyr::rename_with(~gsub(.x,
                           pattern = '\\s|\\.(T|t)rim',
                           replacement = '')) %>%
  dplyr::rename_with(~gsub(.x,
                           pattern = '^(mut[0-9])$',
                           replacement = 'a\\1')) %>%
  dplyr::rename_with(~gsub(.x,
                           pattern = '^(wt[0-9])$',
                           replacement = 'a\\1')) |>
  dplyr::relocate(awt1, .after = p10wtE) |>
  dplyr::relocate(awt3, .before = e13mu1) |>
  dplyr::relocate(awt4, .before = e13mu1) |>
  dplyr::relocate(awt8, .before = e13mu1) |>
  dplyr::relocate(contains('amut'), .after = p10mu4) %>%
  dplyr::rename_with(~gsub(.x, pattern = 'amut([0-9])', replacement = 'amu\\1'))
```

```{r Source the GO_INFO_fn and pass e13 DEGs through it, include = T, message = F, warning = F, echo = F}
source('path/to/GO_Analysis_Functions.R')
GO_INFO_fn(list_of_interest = e13_DRG |>
             dplyr::filter(AdjP <= 0.05) |>
             dplyr::filter(!grepl(GeneID, # remove rRNAs, mitochondrial tRNAs, pseudogenes
                       pattern = 'Rps.+.?$|RP.+.?$|Rpl.+.?$|MRPL.+.?$|Mrpl.+.?$|MRPS.+.?$|Mrps.+.?$|.*Rik.+$|.*Rik$|Gm.+.?$|^[A-Z]+[A-Z].+.?$|^[0-9]+.+.?$|mt.+.?$')
                )|>
             dplyr::select(GeneID) |> unlist() |> as.character(),
           species = 'mouse')
```

```{r Find the DEGs in the Neuron Differentiation GO Term, include = T, message = F, warning = F, echo = T}
neuron_differentiation_DEGs <- GO_INFO_by_TERM_df |>
  dplyr::filter(GO_Term == 'neuron differentiation') |>
  dplyr::select(Gene_IDs_from_List) |>
  unlist() |>
  as.character() %>%
  str_split(., pattern = ';') |>
  unlist()

eDRG <- e13_DRG |>
  dplyr::mutate(devStage = 'e13', .before = 1) |> 
  dplyr::full_join(p0_DRG |>
                     dplyr::mutate(devStage = 'p0', .before = 1) |>
                     dplyr::rename(Base.Mean = Base.mean) |>
                     dplyr::rename(Pval = PVal)) |>
  dplyr::full_join(p10_DRG |>
                     dplyr::mutate(devStage = 'p10', .before = 1) |>
                     dplyr::rename(Base.Mean = Base.mean) |>
                     dplyr::rename(Pval = PVal)) |>
  dplyr::filter(!is.na(AdjP)) |> # remove any NAs in p-value columns
  dplyr::filter(!grepl(GeneID, # remove rRNAs, mitochondrial tRNAs, pseudogenes
                       pattern = 'Rps.+.?$|RP.+.?$|Rpl.+.?$|MRPL.+.?$|Mrpl.+.?$|MRPS.+.?$|Mrps.+.?$|.*Rik.+$|.*Rik$|Gm.+.?$|^[A-Z]+[A-Z].+.?$|^[0-9]+.+.?$|mt.+.?$')
                ) |>
  dplyr::mutate(goi = case_when(AdjP <= 0.05 &
                                  grepl(GeneID,
                                        pattern = 'Wnt.+$|Pou.+$|Fzd.+$|Neuro.+$|Mapk.+$')
                                ~ "Neuron Differentiation DEG",
                                AdjP <= 0.05 &
                                  !grepl(GeneID,
                                         pattern = 'Wnt.+$|Pou.+$|Fzd.+$|Neuro.+$|Mapk.+$')
                                ~ "DEG",
                                TRUE ~ "Non-DEG"))
```


# Plot Embryonic DRG DEA Results {.tabset .tabset-pills .tabset-fade}

#### **Volcano**

```{r Plot eDRG Results, include = T, warning = F, message = F, echo = T, fig.width=14}
ggplot(data = eDRG) +
  geom_point(data = subset(eDRG, goi == 'Non-DEG'),
             aes(x = log2.FC., y = -log10(AdjP), color = goi), alpha = 0.3) +
  geom_point(data = subset(eDRG, goi == 'DEG'),
             aes(x = log2.FC., y = -log10(AdjP), color = goi), alpha = 0.5) +
  geom_point(data = subset(eDRG, goi == 'Neuron Differentiation DEG'),
             aes(x = log2.FC., y = -log10(AdjP), color = goi), alpha = 0.5) +
  labs(title = expression(
    bold(paste("Differential Expression Analysis (DEA) of ",
               italic("Tmem184b")^bold("GT/GT"),
               " eDRG mRNA Exp. Changes"))),
       x = bquote(~ log[2] ~ 'fold change (relative to WT mRNA expression)'),
       y = bquote(~ -log[10] ~ '(Adj.P-value)')
       ) +
  theme_bw() +
  theme(plot.title = element_text(face = 'bold', size = 13),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.ticks = element_line(linetype = 'solid'),
        legend.text = element_text(face = 'bold', size = 12),
        legend.title = element_text(face = 'bold', size = 12),
        strip.background = element_rect(color = 'black', fill = 'white'),
        strip.text = element_text(face = 'bold', size = 13)) +
  geom_hline(yintercept = -log10(0.05), color = 'firebrick', linetype = 'dashed') +
  scale_color_manual(values = c('darkgoldenrod4', 'navy', 'gray48')) +
  geom_text(aes(label = 'FDR\n\u2264 0.05'),
            x = -5,
            y = -log10(0.05),
            size = 5,
            color = 'firebrick',
            show.legend = FALSE) +
  facet_grid(~devStage)

```

#### **MA**

```{r Plot eDRG MA plots, include = T, message = F, warning = F, echo = T, fig.width = 14}
ggplot(data = eDRG) +
  geom_point(data = subset(eDRG, goi == 'Non-DEG'),
             aes(x = log(Base.Mean), y = log2.FC., color = goi), alpha = 0.3) +
  geom_point(data = subset(eDRG, goi == 'DEG'),
             aes(x = log(Base.Mean), y = log2.FC., color = goi), alpha = 0.5) +
  geom_point(data = subset(eDRG, goi == 'Neuron Differentiation DEG'),
             aes(x = log(Base.Mean), y = log2.FC., color = goi), alpha = 1) +
  labs(title = expression(
    bold(paste("Differential Expression Analysis (DEA) of ",
               italic("Tmem184b")^bold("GT/GT"),
               " eDRG mRNA Exp. Changes"))),
       x = bquote(~ ln ~ '(Mean of TPM)'),
       y = bquote(~ -log[2] ~ 'fold change (relative to WT mRNA expression)')
       ) +
  theme_bw() +
  theme(plot.title = element_text(face = 'bold', size = 13),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.ticks = element_line(linetype = 'solid'),
        legend.text = element_text(face = 'bold', size = 12),
        legend.title = element_text(face = 'bold', size = 12),
        strip.background = element_rect(color = 'black', fill = 'white'),
        strip.text = element_text(face = 'bold', size = 13)) +
  geom_hline(yintercept = 0, color = 'black', linetype = 'solid') +
  scale_color_manual(values = c('darkgoldenrod4', 'navy', 'gray48')) +
  scale_fill_manual(values = c('darkgoldenrod4', 'navy', 'gray48')) +
  facet_grid(~devStage)
```


#### **Plot Embryonic DRG Heatmap (Paper Fig 5C)**

Generate the Z-scored normalized counts matrix

```{r Paper eDRG Heatmap, include = T, message = F, warning = F, echo = T, fig.width = 12}
Find_Row_Z(Expression_Profile = TPM |> dplyr::select(c(GeneID, dplyr::starts_with('e13'))))
```

```{r Plot Paper eDRG Heatmap, include = T, message = F, warning = F, echo = T, fig.width = 12}
  ## Create a list of gene names ordered by euclidean distance by which to re-order the dataframe
Euclid_dist_order = hclust(dist(Z[,c(2:7)], method = "euclidean"))$order
  ## The names (not the numbers)
Euclid_dist_ord_Genes = c(Z$GeneID[Euclid_dist_order])

  ## Transform again to order by clusters (Euclidean distances)
Za = Z |>
  dplyr::mutate(GeneID = factor(GeneID, levels = Euclid_dist_ord_Genes)) |>
  dplyr::arrange(GeneID)


temp = vector()
for( i in 1:length(neuron_differentiation_DEGs)){
  temp[i] = which(Euclid_dist_ord_Genes == neuron_differentiation_DEGs[i])
}
diff_index = temp
#diff_index

pheatmap(mat = Za[c(diff_index),2:7],
         color = colorRampPalette(c("navy", "white", "darkgoldenrod4"))((50)),
         clustsering_distance_rows = "euclidean",
         angle_col = 0,
         treeheight_row = 35,
         treeheight_col = 9,
         cutree_rows = 4,
         labels_row = Euclid_dist_ord_Genes[c(diff_index)])
```


#### **Paper Fig 5B**

The GeneOntology.org db has changed since 2018 when we did our original analyses, thus some terms are different in terms of their significance.

+ The terms in the paper have each been filtered directly

```{r Retro Paper Pathway fig, include = T, message = F, warning = F, echo = F, fig.height = 10, fig.width=12}
eDRG_GO_results |>
  dplyr::filter(devStage == 'e13', GO_Type == 'BP') |>
  dplyr::arrange(FDR, desc(fold.Enrichment)) |> 
  dplyr::mutate(sig = case_when(FDR <= 0.001 ~ "***",
                                between(FDR, 0.001, 0.0499) ~ "**",
                                between(FDR, 0.01, 0.05) ~ "*",
                                TRUE ~ "NS"), .after = FDR) |>
  dplyr::filter(GO_Term %in%
                  c('axonogenesis',
                    'nervous system development',
                    'regulation of calcium-dependent activation of synaptic vesicle fusion', # re-categorized from extant 'reg. of calcium-dependent exocytosis'
                    'axon development', 'chromosome localization',
                    'neuron differentiation', 'ganglion development',
                    'G1/S transition of mitotic cell cycle', # re-categorized from G1/S transition
                    'epithelial cell-cell adhesion', # re-categorized from epithelial cell adhesion
                    'generation of neurons' # re-categorized from neuron generation
                               )) |>
  dplyr::mutate(GO_Term = case_when(grepl(GO_Term,
                                          pattern = 'calcium-dependent')
                                    ~ 'regulation of calcium-dependent\nactivation of synaptic vesicle fusion',
                                    TRUE ~ GO_Term)) |> 
  ggplot(aes(x = forcats::fct_rev(forcats::fct_infreq(GO_Term, FDR)),
             y = -log10(FDR),
             color = -log10(FDR),
             fill = -log10(FDR))) +
  geom_bar(stat = 'identity') +
  theme_bw()+
  theme(plot.title = element_text(face = 'bold', size = 13),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text.y = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.text.x = element_text(face = 'bold',
                                   size = 12,
                                   vjust = 0.5,
                                   angle = 90,
                                   hjust = 1),
        axis.ticks = element_line(linetype = 'solid'),
        legend.position = 'none',
        # legend.text = element_text(face = 'bold', size = 12),
        # legend.title = element_text(face = 'bold', size = 12),
        panel.grid = element_blank()) +
  labs(title = expression(bold(paste("GO Analysis (GSEA) of ",
                                     italic("Tmem184b")^bold("GT/GT"),
                                     " e13 DRG mRNA Exp. Changes"))),
       x = 'GO Biological Process',
       y = bquote(~ -log[10] ~ '(Adj.P-value)')
       ) +
  geom_text(aes(x = GO_Term,
                y = -log10(FDR)+0.75,
                label = sig,
                fontface = 'bold'),
            size = 6.5,
            color = 'black',
            vjust = 0.5,
            hjust = 0.5,
            alpha = 1)
```

Here is the current analysis of the same data

```{r Current Paper Pathway fig, include = T, message = F, warning = F, echo = T, fig.height = 10, fig.width=12}
eDRG_GO_results |>
  dplyr::filter(devStage == 'e13', GO_Type == 'BP') |>
  dplyr::arrange(FDR, desc(fold.Enrichment)) |> 
  dplyr::mutate(sig = case_when(FDR <= 0.001 ~ "***",
                                between(FDR, 0.001, 0.0499) ~ "**",
                                between(FDR, 0.01, 0.05) ~ "*",
                                TRUE ~ "NS"), .after = FDR) |>
  dplyr::slice(c(1:15)) |> 
  ggplot(aes(x = forcats::fct_rev(forcats::fct_infreq(GO_Term, FDR)),
             y = -log10(FDR),
             color = -log10(FDR),
             fill = -log10(FDR))) +
  geom_bar(stat = 'identity') +
  theme_bw()+
  theme(plot.title = element_text(face = 'bold', size = 13),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text.y = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.text.x = element_text(face = 'bold',
                                   size = 12,
                                   vjust = 0.5,
                                   angle = 90,
                                   hjust = 1),
        axis.ticks = element_line(linetype = 'solid'),
        legend.text = element_text(face = 'bold', size = 12),
        legend.title = element_text(face = 'bold', size = 12),
        panel.grid = element_blank()) +
  labs(title = expression(bold(paste("GO Analysis (GSEA) of ",
                                     italic("Tmem184b")^bold("GT/GT"),
                                     " e13 DRG mRNA Exp. Changes"))),
       x = 'GO Biological Process',
       y = bquote(~ -log[10] ~ '(Adj.P-value)')
       ) +
  geom_text(aes(x = GO_Term,
                y = -log10(FDR)+0.75,
                label = sig,
                fontface = 'bold'),
            size = 6.5,
            color = 'black',
            vjust = 0.5,
            hjust = 0.5,
            alpha = 1)
```

