---
title: "Itch GSEA"
author: "Erik Larsen"
date: "2025-02-10"
output:
  html_document:
    code_folding: hide
    toc: TRUE
---

# Overview

This markdown was developed to formally "modernize" the gene set enrichment analysis of bulk RNA-seq data
for **Dr. Martha Bhattacharya's lab** at the **University of Arizona** for the [Itch paper](https://journals.lww.com/pain/abstract/2022/05000/transmembrane_protein_tmem184b_is_necessary_for.18.aspx)

## Set up Environment {.tabset .tabset-pills .tabset-fade}

#### **Load packages**

```{r Package Load, include = T, message = F, warning = F, echo = T}
packages <- c('BiocManager', 'tidyverse', 'stringr', 'GO.db', 'org.Mm.eg.db', 'forcats', 'ggplot2', 'ggrepel')
for ( package in packages ) {
  if (!require(package, character.only = T)) {
    install.packages(package)
    library(package, character.only = T)
  }
}
```

# Read in and Prep aDRG Data {.tabset .tabset-pills .tabset-fade}

#### **Read in aDRG Data**

Not shown; variable names:

+ `aDRG` = adult mouse DRG DEA results
  + `GO_{category}_results` = gene.ontology gene set enrichment analysis results

```{r Read in aDRG RNAseq Results, include = T, message = F, warning = F, echo = F}
aDRG = read.csv("path/to/DESeq2 Expression Results.csv")

aDRG_GO_BP_results <- read.table('path/to/aDRG_GO_analysis_BP.txt', sep = '\t', skip = 11, header = T, quote = '')
aDRG_GO_CC_results <- read.table('path/to/aDRG_GO_analysis_CC.txt', sep = '\t', skip = 11, header = T, quote = '')
aDRG_GO_MF_results <- read.table('path/to/aDRG_GO_analysis_MF.txt', sep = '\t', skip = 11, header = T, quote = '')

aDRG_PANTHER_Pathways <- read.table('path/to/aDRG_PANTHER_Pathways.txt', sep = '\t', skip = 11, header = T, quote = '') |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) %>% # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) %>% # rm appended/excess column name strings
  dplyr::rename(Pathway.Size = REFLIST21836) |>
  dplyr::rename(Overlap = `364`) %>%
  dplyr::mutate(across(c(2:4,6:8), ~ as.numeric(.x))) |> # co-erce to numeric for operations
  tidyr::separate(PANTHER.Pathways, into = c("PANTHER.Pathway", "Panther.Pathway.ID"), sep = '\\s\\(P') |> # split to preserve names & IDs
  dplyr::mutate(Panther.Pathway.ID = paste0('P', gsub(Panther.Pathway.ID, pattern = '\\)', replacement = '')))
```

#### **aDRG GO Results Wrangling**

```{r Join aDRG GO results, include = T, message = F, warning = F, echo = T}
aDRG_GO_BP_results <- aDRG_GO_BP_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.biological.process.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `364`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "BP", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

aDRG_GO_CC_results <- aDRG_GO_CC_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.cellular.component.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `364`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "CC", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

aDRG_GO_MF_results <- aDRG_GO_MF_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.molecular.function.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `364`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "MF", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

aDRG_GO_results <- aDRG_GO_BP_results |>
  dplyr::full_join(aDRG_GO_CC_results) |>
  dplyr::full_join(aDRG_GO_MF_results)

rm(list = ls()[which(grepl(ls(), pattern = 'aDRG_GO_[A-Z]') == T)])
```

# Plot aDRG Results

#### **GO GSEA**

```{r Plot aDRG GO GSEA results, include = T, message = F, warning = F, echo = T, fig.width=12}
aDRG_GO_results |>
  dplyr::filter(!is.na(GO_Term)) |>
  dplyr::mutate(fold.Enrichment = as.numeric(fold.Enrichment)) |>
  dplyr::arrange(desc(fold.Enrichment), FDR) |>
  dplyr::filter(!grepl(GO_Term, pattern = "Unclassified", ignore.case = T)) |>
  dplyr::mutate(sig = case_when(FDR <= 0.001 ~ "***",
                                between(FDR, 0.001, 0.0499) ~ "**",
                                between(FDR, 0.01, 0.05) ~ "*",
                                TRUE ~ "NS"), .after = FDR
  ) |>
  dplyr::filter(sig != 'NS') |>
  dplyr::slice(c(1:30)) |> 
  ggplot(aes(x = fold.Enrichment,
             y = forcats::fct_rev(forcats::fct_infreq(GO_Term, fold.Enrichment)),
             color = GO_Type,
             fill = GO_Type,
             size = Overlap,
             alpha = -log10(FDR))) +
  geom_point() +
  labs(title = expression(bold(paste("GO Analysis (GSEA) of ",
                                     italic("Tmem184b")^bold("GT/GT"),
                                     " DRG mRNA Exp. Changes")
                               )
                          ),
       # x = bquote(~ -log[10] ~ '(Adj.P-value)'),
       x = 'fold enrichment',
       y = 'GO Term',
       color = 'GO\nCategory',
       fill = 'GO\nCategory',
       size = '# Genes in List',
       # alpha = 'fold\nEnrichment'
       alpha = bquote(~ -log[10] ~ '(Adj.P-value)')
       ) +
  # geom_vline(xintercept = -log10(0.05), linetype = 'dashed', color = 'firebrick') +
  theme_bw() +
  theme(plot.title = element_text(face = 'bold', size = 13),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.ticks = element_line(linetype = 'solid'),
        legend.text = element_text(face = 'bold', size = 12),
        legend.title = element_text(face = 'bold', size = 12)) +
  guides(fill = guide_legend(override.aes = list(size = 5,
                                                 shape = 19,
                                                 fill = c('navy',
                                                          'darkgoldenrod3',
                                                          'firebrick'))),
         alpha = guide_legend(override.aes = list(size = 5, shape = 19)),
         size = guide_legend(override.aes = list(shape = 19))) +
  scale_color_manual(values = c('navy', 'darkgoldenrod3', 'firebrick')) +
  scale_fill_manual(values = c('navy', 'darkgoldenrod3', 'firebrick')) +
  geom_text(aes(x = fold.Enrichment+ 0.2,
                y = GO_Term,
                color = GO_Type,
                label = sig,
                fontface = 'bold'),
            size = 5.5,
            angle = 90,
            vjust = 1,
            alpha = 1)
```

#### **PANTHER Pathways GSEA**

```{r Plot aDRG PANTHER Pathways GSEA, include = T, message = F, warning = F, echo = T, fig.width = 12}
aDRG_PANTHER_Pathways |>
  dplyr::mutate(sig = case_when(FDR <= 0.001 ~ "***",
                                between(FDR, 0.001, 0.0499) ~ "**",
                                between(FDR, 0.01, 0.05) ~ "*",
                                TRUE ~ "NS"), .after = FDR
  ) |>
  dplyr::filter(sig != 'NS') |>
  dplyr::arrange(desc(fold.Enrichment), FDR) |> 
  dplyr::slice(c(1:10)) |> 
  ggplot(aes(x = -log10(FDR),
             y = forcats::fct_rev(forcats::fct_infreq(PANTHER.Pathway, -log10(FDR))),
             color = -log10(FDR),
             fill = -log10(FDR))) +
  geom_bar(stat = 'identity') +
  labs(title = expression(bold(paste("PANTHER Pathway Analysis (GSEA) of ",
                                     italic("Tmem184b")^bold("GT/GT"),
                                     " aDRG mRNA Exp. Changes")
                               )
                          ),
       x = bquote(~ -log[10] ~ '(Adj.P-value)'),
       y = 'PANTHER Pathway',
       alpha = bquote(~ -log[10] ~ '(Adj.P-value)')
       ) +
  geom_vline(xintercept = -log10(0.05), linetype = 'dashed', color = 'firebrick') +
  theme_bw() +
  theme(plot.title = element_text(face = 'bold', size = 13),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.ticks = element_line(linetype = 'solid'),
        legend.position = 'none'
        ) +
  # guides(alpha = guide_legend(override.aes = list(size = 5, shape = 19)),
  #        size = guide_legend(override.aes = list(shape = 19))
  #        ) +
  geom_text(aes(label = 'FDR\n\u2264 0.05'),
            size = 5,
            color = 'firebrick',
            x = 3,
            y = 2,
            show.legend = FALSE) +
  geom_text(aes(x = -log10(FDR)+0.1,
                label = sig,
                fontface = 'bold'),
            size = 6.5,
            color = 'black',
            angle = 90,
            vjust = 1,
            alpha = 1)
```


# Read in and Prep eDRG Data {.tabset .tabset-pills .tabset-fade}

#### **Read in eDRG Data**

Not shown; variable names:

+ `({dev_stage_group})_DRG` = DRG DEA results
  + `({dev_stage_group})_GO_{category}_results` = gene.ontology results

```{r Read in eDRG Results, include = T, warning = F, message = F, echo = F}
e13_DRG <- read.csv("path/to/DESeq2_result_file_on_GT_e13_PARA.csv", header = F)
colnames(e13_DRG) = colnames(aDRG)[c(1:7)]

write.table(e13_DRG |> dplyr::filter(AdjP <= 0.01) |> dplyr::select(GeneID) |> unlist() |> as.character(), 'path/to/e13_DEGs.csv', row.names = F, col.names = F, sep = ',')

e13_DRG_GO_BP_results <- read.table('C:/Users/Erik/Desktop/Programming/R/Bio/Itch/e13_DRG_GO_analysis_BP.txt', sep = '\t', skip = 11, header = T, quote = '')
e13_DRG_GO_CC_results <- read.table('C:/Users/Erik/Desktop/Programming/R/Bio/Itch/e13_DRG_GO_analysis_CC.txt', sep = '\t', skip = 11, header = T, quote = '')
e13_DRG_GO_MF_results <- read.table('C:/Users/Erik/Desktop/Programming/R/Bio/Itch/e13_DRG_GO_analysis_MF.txt', sep = '\t', skip = 11, header = T, quote = '')
e13_PANTHER_Pathways <- read.table('C:/Users/Erik/Desktop/Programming/R/Bio/Itch/e13_PANTHER_Pathways.txt', sep = '\t', skip = 11, header = T, quote = '')

p0_DRG <- read.csv("path/to/DESeq2_result_file_on_GT_p0_mu234_wtBCD_PARA.csv", header = T)

write.table(p0_DRG |> dplyr::filter(AdjP <= 0.01) |> dplyr::select(GeneID) |> unlist() |> as.character(), 'path/to/p0_DEGs.csv', row.names = F, col.names = F, sep = ',')

p0_DRG_GO_BP_results <- read.table('path/to/p0_DRG_GO_analysis_BP.txt', sep = '\t', skip = 11, header = T, quote = '')
p0_DRG_GO_CC_results <- read.table('path/to/p0_DRG_GO_analysis_CC.txt', sep = '\t', skip = 11, header = T, quote = '')
p0_DRG_GO_MF_results <- read.table('path/to/p0_DRG_GO_analysis_MF.txt', sep = '\t', skip = 11, header = T, quote = '')
p0_PANTHER_Pathways <- read.table('path/to/p0_PANTHER_Pathways.txt', sep = '\t', skip = 11, header = T, quote = '')

p10_DRG <- read.csv("path/to/DESeq2_result_file_on_GT_p10_mu123_wt1CE_PARA.csv", header = T)

write.table(p10_DRG |> dplyr::filter(AdjP <= 0.01) |> dplyr::select(GeneID) |> unlist() |> as.character(), 'path/to/p10_DEGs.csv', row.names = F, col.names = F, sep = ',')

p10_DRG_GO_BP_results <- read.table('path/to/p10_DRG_GO_analysis_BP.txt', sep = '\t', skip = 11, header = T, quote = '')
p10_DRG_GO_CC_results <- read.table('path/to/p10_DRG_GO_analysis_CC.txt', sep = '\t', skip = 11, header = T, quote = '')
p10_DRG_GO_MF_results <- read.table('path/to/p10_DRG_GO_analysis_MF.txt', sep = '\t', skip = 11, header = T, quote = '')
p10_PANTHER_Pathways <- read.table('patht/to/p10_PANTHER_Pathways.txt', sep = '\t', skip = 11, header = T, quote = '')
```

#### **e13 DRG GO Results Wrangling**

```{r Join e13 DRG GO results, include = T, message = F, warning = F, echo = T}
e13_DRG_GO_BP_results <- e13_DRG_GO_BP_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.biological.process.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `1411`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "BP", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

e13_DRG_GO_CC_results <- e13_DRG_GO_CC_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.cellular.component.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `1411`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "CC", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

e13_DRG_GO_MF_results <- e13_DRG_GO_MF_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.molecular.function.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `1411`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "MF", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

e13_DRG_GO_results <- e13_DRG_GO_BP_results |>
  dplyr::full_join(e13_DRG_GO_CC_results) |>
  dplyr::full_join(e13_DRG_GO_MF_results) |>
  dplyr::mutate(devStage = 'e13', .before = 1)
```

#### **p0 DRG GO Results Wrangling**

```{r Join p0 DRG GO results, include = T, message = F, warning = F, echo = T}
p0_DRG_GO_BP_results <- p0_DRG_GO_BP_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.biological.process.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `3078`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "BP", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

p0_DRG_GO_CC_results <- p0_DRG_GO_CC_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.cellular.component.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `3078`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "CC", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

p0_DRG_GO_MF_results <- p0_DRG_GO_MF_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.molecular.function.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `3078`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "MF", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

p0_DRG_GO_results <- p0_DRG_GO_BP_results |>
  dplyr::full_join(p0_DRG_GO_CC_results) |>
  dplyr::full_join(p0_DRG_GO_MF_results) |>
  dplyr::mutate(devStage = 'p0', .before = 1)
```

#### **p10 DRG GO Results Wrangling**

```{r Join p10 DRG GO results, include = T, message = F, warning = F, echo = T}
p10_DRG_GO_BP_results <- p10_DRG_GO_BP_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.biological.process.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `107`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "BP", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

p10_DRG_GO_CC_results <- p10_DRG_GO_CC_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.cellular.component.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `107`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "CC", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

p10_DRG_GO_MF_results <- p10_DRG_GO_MF_results |>
  dplyr::rename_with(~gsub(.x, pattern = '\\.\\.', replacement = '')) |> # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x, pattern = 'Mus.musculus.|upload_1|\\.$', replacement = '')) |> # remove excess strings
  dplyr::rename(GO_Term = GO.molecular.function.complete) |> # rename the BP column
  dplyr::rename(GO_Term_Size = REFLIST21836) |> # rename the GO Term Size column
  dplyr::rename(Overlap = `107`) |> # rename the overlap column
  dplyr::mutate(across(c(2:4,7,8), ~as.numeric(.x))) |> # convert to numeric
  tidyr::separate(GO_Term, into = c("GO_Term", "GO_Term_ID"), sep = '\\s\\(GO:') |> # split the Term ID from the Term
  dplyr::mutate(GO_Term_ID = paste0('GO:', gsub(GO_Term_ID, pattern = '\\)', replacement = ''))) |>
  dplyr::mutate(GO_Type = "MF", .after = GO_Term_ID) |> # create the category column
  dplyr::mutate(fold.Enrichment = str_extract(fold.Enrichment, pattern = '[:alnum:]+') |> as.numeric(fold.Enrichment)) |> # convert the FE column to numeric
  dplyr::arrange(FDR, desc(fold.Enrichment)) # sort by the highest FE and lowest adjusted p-value

p10_DRG_GO_results <- p10_DRG_GO_BP_results |>
  dplyr::full_join(p10_DRG_GO_CC_results) |>
  dplyr::full_join(p10_DRG_GO_MF_results) |>
  dplyr::mutate(devStage = 'p10', .before = 1)

rm(list = ls()[which(grepl(ls(), pattern = '_DRG_GO_[A-Z]|aDRG_GO_[A-Z]') == T)])
```

# Plot Embryonic DRG GO Results {.tabset .tabset-pills .tabset-fade}

```{r Plot eDRG Results, include = T, warning = F, message = F, echo = T, fig.width=14}
eDRG_GO_results <- e13_DRG_GO_results |>
  dplyr::full_join(p0_DRG_GO_results) |>
  dplyr::full_join(p10_DRG_GO_results)

eDRG_GO_results |>
  dplyr::filter(!is.na(GO_Term)) |>
  dplyr::mutate(fold.Enrichment = as.numeric(fold.Enrichment)) |>
  dplyr::arrange(desc(fold.Enrichment), FDR) |>
  dplyr::filter(!grepl(GO_Term, pattern = "Unclassified", ignore.case = T)) |>
  dplyr::mutate(sig = case_when(FDR <= 0.001 ~ "***",
                                between(FDR, 0.001, 0.0499) ~ "**",
                                between(FDR, 0.01, 0.05) ~ "*",
                                TRUE ~ "NS"), .after = FDR
  ) |>
  dplyr::filter(sig != 'NS') |>
  dplyr::slice(c(1:30)) |> 
  dplyr::mutate(GO_Term = case_when(grepl(GO_Term, pattern = 'adaptive immune response based on somatic') ~ 
                                      'adaptive immune response based on somatic recombination of\nimmune receptors built from immunoglobulin superfamily domains',
                                    TRUE ~ GO_Term)) |>
  ggplot(aes(x = fold.Enrichment,
             y = forcats::fct_rev(forcats::fct_infreq(GO_Term, fold.Enrichment)),
             color = GO_Type,
             fill = GO_Type,
             shape = devStage,
             size = Overlap,
             alpha = -log10(FDR))) +
  geom_point(aes(shape = devStage)) +
  labs(title = expression(bold(paste("GO Analysis (GSEA) of ", italic("Tmem184b")^bold("GT/GT"), " eDRG mRNA Exp. Changes"))),
       x = 'fold Enrichment',
       # x = bquote(~ -log[10] ~ '(Adj.P-value)'),
       y = 'GO Term',
       color = 'GO\nCategory',
       fill = 'GO\nCategory',
       shape = 'Dev Stage',
       size = 'Overlap',
       # alpha = 'fold\nEnrichment'
       alpha = '-log10(FDR)'
       ) +
  # geom_vline(xintercept = -log10(0.05), linetype = 'dashed', color = 'firebrick') +
  theme_bw() +
  theme(plot.title = element_text(face = 'bold', size = 13),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.ticks = element_line(linetype = 'solid'),
        legend.text = element_text(face = 'bold', size = 12),
        legend.title = element_text(face = 'bold', size = 12),
        strip.background = element_rect(color = 'black', fill = 'white'),
        strip.text = element_text(face = 'bold', size = 13)) +
  guides(shape = guide_legend(override.aes = list(size = 4, shape = c(19, 17))),
         fill = guide_legend(override.aes = list(size = 4, shape = 19, color = c('navy', 'darkgoldenrod3', 'firebrick'))),
         alpha = guide_legend(override.aes = list(size = 5, shape = 19)),
         size = guide_legend(override.aes = list(shape = 19))
         # size = 'none'
         ) +
  scale_color_manual(values = c('navy', 'darkgoldenrod3', 'firebrick')) +
  scale_fill_manual(values = c('navy', 'darkgoldenrod3', 'firebrick')) +
  # geom_text(aes(label = 'FDR\n\u2264 0.05'),
  #           size = 5,
  #           color = 'firebrick',
  #           x = 2,
  #           y = 4,
  #           show.legend = FALSE) +
  geom_text(aes(x = fold.Enrichment+1,
                label = sig,
                fontface = 'bold',
                color = GO_Type),
            size = 6.5,
            # color = 'black',
            angle = 90,
            vjust = 1,
            alpha = 1) +
  facet_grid(~devStage)
```

#### **Plot Embryonic DRG PANTHER Pathway Results**

```{r Plot eDRG PANTHER Results, include = T, warning = F, message = F, echo = T, fig.width=12}
eDRG_PANTHER_Pathways <- e13_PANTHER_Pathways |>
  dplyr::mutate(devStage = 'e13', .before = 1) |>
  dplyr::rename_with(~gsub(.x,
                           pattern = '\\.\\.',
                           replacement = '')) %>% # remove the .'s  from the GO BP column
  dplyr::rename_with(~gsub(.x,
                           pattern = 'Mus.musculus.|upload_1|\\.$',
                           replacement = '')) |> # rm appended/excess column name strings
  dplyr::rename(Pathway.Size = REFLIST21836) |>
  dplyr::rename(Overlap = `1411`) %>%
  dplyr::mutate(across(c(3:5,7:9), ~ as.numeric(.x))) |> # co-erce to numeric for operations
  tidyr::separate(PANTHER.Pathways,
                  into = c("PANTHER.Pathway", "Panther.Pathway.ID"),
                  sep = '\\s\\(P') |> # split to preserve names & IDs
  dplyr::mutate(Panther.Pathway.ID = paste0('P',
                                            gsub(Panther.Pathway.ID,
                                                 pattern = '\\)',
                                                 replacement = ''))) |>

  dplyr::full_join(p0_PANTHER_Pathways |>
                     dplyr::mutate(devStage = 'p0', .before = 1) %>%
                     dplyr::rename_with(~gsub(.x,
                                              pattern = '\\.\\.',
                                              replacement = '')) %>% # remove the .'s  from the GO BP column
                     dplyr::rename_with(~gsub(.x,
                                              pattern = 'Mus.musculus.|upload_1|\\.$',
                                              replacement = '')) |>  # rm appended/excess column name strings
                     dplyr::rename(Pathway.Size = REFLIST21836) |>
                     dplyr::rename(Overlap = `3078`) %>%
                     dplyr::mutate(across(c(3:5,7:9), ~ as.numeric(.x))) |> # co-erce to numeric for operations
                     tidyr::separate(PANTHER.Pathways,
                                     into = c("PANTHER.Pathway", "Panther.Pathway.ID"),
                                     sep = '\\s\\(P') |> # split to preserve names & IDs
                     dplyr::mutate(Panther.Pathway.ID = paste0('P',
                                                               gsub(Panther.Pathway.ID,
                                                                    pattern = '\\)',
                                                                    replacement = '')))
                   ) |>
  dplyr::full_join(p10_PANTHER_Pathways |>
                     dplyr::mutate(devStage = 'p10', .before = 1) %>%
                     dplyr::rename_with(~gsub(.x,
                                              pattern = '\\.\\.',
                                              replacement = '')) %>% # remove the .'s  from the GO BP column
                     dplyr::rename_with(~gsub(.x,
                                              pattern = 'Mus.musculus.|upload_1|\\.$',
                                              replacement = '')) |> # rm appended/excess column name strings
                     dplyr::rename(Pathway.Size = REFLIST21836) |>
                     dplyr::rename(Overlap = `107`) %>%
                     dplyr::mutate(across(c(3:5,7:9), ~ as.numeric(.x))) |> # co-erce to numeric for operations
                     tidyr::separate(PANTHER.Pathways,
                                     into = c("PANTHER.Pathway", "Panther.Pathway.ID"),
                                     sep = '\\s\\(P') |> # split to preserve names & IDs
                     dplyr::mutate(Panther.Pathway.ID = paste0('P',
                                                               gsub(Panther.Pathway.ID,
                                                                    pattern = '\\)',
                                                                    replacement = '')))
                   ) |>
  dplyr::filter(!grepl(PANTHER.Pathway, pattern = 'Unclassified')) |>
  dplyr::arrange(desc(fold.Enrichment), FDR) |> # sort by the highest FE and lowest adjusted p-value
  
  dplyr::mutate(sig = case_when(FDR <= 0.001 ~ "***",
                                between(FDR, 0.001, 0.0499) ~ "**",
                                between(FDR, 0.01, 0.05) ~ "*",
                                TRUE ~ "NS"), .after = FDR
  )

eDRG_PANTHER_Pathways |>
  dplyr::filter(sig != 'NS') |>
  dplyr::arrange(desc(fold.Enrichment), FDR) |> 
  dplyr::slice(c(1:30)) |> 
  # dplyr::group_by(devStage, PANTHER.Pathway) |>
  ggplot(aes(x = fold.Enrichment,
             y = forcats::fct_rev(forcats::fct_infreq(PANTHER.Pathway, fold.Enrichment)),
             color = devStage,
             fill = devStage,
             size = Overlap,
             alpha = -log10(FDR))) +
  geom_point() +
  # geom_bar(stat = 'identity') +
  labs(title = expression(bold(paste("GO Analysis (GSEA) of ",
                                     italic("Tmem184b")^bold("GT/GT"),
                                     " eDRG mRNA Exp. Changes"))),
       x = 'fold Enrichment',
       # x = bquote(~ -log[10] ~ '(Adj.P-value)'),
       y = 'PANTHER Pathway',
       color = 'devStage',
       fill = 'devStage',
       size = 'Overlap',
       # alpha = 'fold\nEnrichment'
       alpha = '-log10(FDR)'
       ) +
  # geom_vline(xintercept = -log10(0.05), linetype = 'dashed', color = 'firebrick') +
  theme_bw() +
  theme(plot.title = element_text(face = 'bold', size = 13),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.ticks = element_line(linetype = 'solid'),
        legend.text = element_text(face = 'bold', size = 12),
        legend.title = element_text(face = 'bold', size = 12),
        strip.background = element_rect(color = 'black', fill = 'white'),
        strip.text = element_text(face = 'bold', size = 12)) +
  guides(fill = guide_legend(override.aes = list(size = 4,
                                                 shape = 19,
                                                 color = c('navy',
                                                           'darkgoldenrod3'))),
         alpha = guide_legend(override.aes = list(size = 5, shape = 19)),
         size = guide_legend(override.aes = list(shape = 19))
         ) +
  scale_color_manual(values = c('navy', 'darkgoldenrod3')) +
  scale_fill_manual(values = c('navy', 'darkgoldenrod3')) +
  # geom_text(aes(label = 'FDR\n\u2264 0.05'),
  #           size = 5,
  #           color = 'firebrick',
  #           x = 2,
  #           y = 4,
  #           show.legend = FALSE) +
  geom_text(aes(x = fold.Enrichment+0.1,
                label = sig,
                fontface = 'bold',
                color = devStage),
            size = 6.5,
            angle = 90,
            vjust = 1,
            alpha = 1) +
  facet_grid(~devStage)
```

#### **Paper Fig 5B**

The GeneOntology.org db has changed since 2018 when we did our original analyses, thus some terms are different in terms of their significance.

+ The terms in the paper have each been filtered directly

```{r Retro Paper Pathway fig, include = T, message = F, warning = F, echo = F, fig.height=7, fig.width=12}
eDRG_GO_results |>
  dplyr::filter(devStage == 'e13', GO_Type == 'BP') |>
  dplyr::arrange(FDR, desc(fold.Enrichment)) |> 
  dplyr::mutate(sig = case_when(FDR <= 0.001 ~ "***",
                                between(FDR, 0.001, 0.0499) ~ "**",
                                between(FDR, 0.01, 0.05) ~ "*",
                                TRUE ~ "NS"), .after = FDR) |>
  dplyr::filter(GO_Term %in% c('axonogenesis',
                               'nervous system development',
                               'regulation of calcium-dependent activation of synaptic vesicle fusion', # re-categorized from extant 'reg. of calcium-dependent exocytosis'
                               'axon development', 'chromosome localization',
                               'neuron differentiation', 'ganglion development',
                               'G1/S transition of mitotic cell cycle', # re-categorized from G1/S transition
                               'epithelial cell-cell adhesion', # re-categorized from epithelial cell adhesion
                               'generation of neurons' # re-categorized from neuron generation
                               )) |>
  dplyr::mutate(GO_Term = case_when(grepl(GO_Term, pattern = 'calcium-dependent') ~ 'regulation of calcium-dependent\nactivation of synaptic vesicle fusion',
                                    TRUE ~ GO_Term)) |> 
  ggplot(aes(x = forcats::fct_rev(forcats::fct_infreq(GO_Term, FDR)), y = -log10(FDR), color = -log10(FDR), fill = -log10(FDR))) +
  geom_bar(stat = 'identity') +
  theme_bw()+
  theme(plot.title = element_text(face = 'bold', size = 13),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text.y = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.text.x = element_text(face = 'bold', size = 12, vjust = 0.5, angle = 90, hjust = 1),
        axis.ticks = element_line(linetype = 'solid'),
        legend.position = 'none',
        # legend.text = element_text(face = 'bold', size = 12),
        # legend.title = element_text(face = 'bold', size = 12),
        panel.grid = element_blank()) +
  labs(title = expression(bold(paste("GO Analysis (GSEA) of ", italic("Tmem184b")^bold("GT/GT"), " e13 DRG mRNA Exp. Changes"))),
       x = 'GO Biological Process',
       y = bquote(~ -log[10] ~ '(Adj.P-value)')
       ) +
  geom_text(aes(x = GO_Term,
                y = -log10(FDR)+0.75,
                label = sig,
                fontface = 'bold'),
            size = 6.5,
            color = 'black',
            vjust = 0.5,
            hjust = 0.5,
            alpha = 1)
```

Here is the current analysis of the same data

```{r Current Paper Pathway fig, include = T, message = F, warning = F, echo = T, fig.height=7, fig.width=12}
eDRG_GO_results |>
  dplyr::filter(devStage == 'e13', GO_Type == 'BP') |>
  dplyr::arrange(FDR, desc(fold.Enrichment)) |> 
  dplyr::mutate(sig = case_when(FDR <= 0.001 ~ "***",
                                between(FDR, 0.001, 0.0499) ~ "**",
                                between(FDR, 0.01, 0.05) ~ "*",
                                TRUE ~ "NS"), .after = FDR) |>
  dplyr::slice(c(1:15)) |> 
  ggplot(aes(x = forcats::fct_rev(forcats::fct_infreq(GO_Term, FDR)), y = -log10(FDR), color = -log10(FDR), fill = -log10(FDR))) +
  geom_bar(stat = 'identity') +
  theme_bw()+
  theme(plot.title = element_text(face = 'bold', size = 13),
        axis.title = element_text(face = 'bold', size = 12),
        axis.text.y = element_text(face = 'bold', size = 12, vjust = 0.5),
        axis.text.x = element_text(face = 'bold', size = 12, vjust = 0.5, angle = 90, hjust = 1),
        axis.ticks = element_line(linetype = 'solid'),
        legend.text = element_text(face = 'bold', size = 12),
        legend.title = element_text(face = 'bold', size = 12),
        panel.grid = element_blank()) +
  labs(title = expression(bold(paste("GO Analysis (GSEA) of ", italic("Tmem184b")^bold("GT/GT"), " e13 DRG mRNA Exp. Changes"))),
       x = 'GO Biological Process',
       y = bquote(~ -log[10] ~ '(Adj.P-value)')
       ) +
  geom_text(aes(x = GO_Term,
                y = -log10(FDR)+0.75,
                label = sig,
                fontface = 'bold'),
            size = 6.5,
            color = 'black',
            vjust = 0.5,
            hjust = 0.5,
            alpha = 1)
```

